Tensile/Tests/unit/test_TensileCreateLibrary.py

```diff
@@ -175,3 +175,103 @@ def test_verifyManifest():
   
     with open("bar.asm", "x"):
       assert TensileCreateLibrary.verifyManifest(manifestFile), "files in manifest should be on disk"
+
+def test_parseGfxArchitectures():
+    # Test `all` architectures
+    archs = 'all'
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    expected = set([archs])
+    assert result == expected, f"arch `{archs}` should parse to {expected} but instead maps to {result}"
+
+    expected = set([item for item in filter(lambda x: x != "all", Common.architectureMap.keys())])
```

Comment:
We should consider manually typing out the expected result. Consider what would happen if we added a new entry and made a typo in the value. - would the test fail? Another potential gotcha is someone inadvertently changes `architectureMap`. This test would continue to pass because the expected value is computed using the state that we are testing.